console-subscriber/src/aggregator/mod.rs

```diff
@@ -320,19 +322,36 @@ impl Aggregator {
             let (tx, rx) = mpsc::channel(buffer);
             let subscription = Watch(tx);
             let now = Some(self.base_time.to_timestamp(Instant::now()));
+
+            let span = id.clone();
+            // XXX(jswrenn): A new thread is spawned solely for the sake of
+            // getting access to the global subscriber.
+            let (trace, updates) = std::thread::spawn(move || {
+                // XXX(jswrenn): Need to gracefully handle the case where the
+                // task span is already closed.
+                consequences::trace(&span, buffer)
+                    .expect("the subscriber isn't a Registry")
+                    .expect("id is already closed")
+            })
+            .join()
+            .unwrap();
```

Comment:
I'd like to find a better alternative to this.

Reply:
I think I found an alternative to this! It's just blocked on https://github.com/tokio-rs/tracing/pull/2247.